sympy/solvers/tests/test_recurr.py

```diff
@@ -184,6 +184,10 @@ def test_rsolve():
             (C0*((sqrt(-a**2 + 1) - 1)/a)**n +
              C1*((-sqrt(-a**2 + 1) - 1)/a)**n)).simplify() == 0
 
+    assert rsolve((k + 1)*y(k), y(k)) is None
+    assert (rsolve((k + 1)*y(k) + (k + 3)*y(k + 1) + (k + 5)*y(k + 2), y(k))
+            is None)
```

Comment:
Docstring of recurr module

> This module is intended for solving recurrences or, in other words,
    difference equations. Currently supported are linear, inhomogeneous
    equations with polynomial or rational coefficients.

I guess they fall in this category. So, input is valid.

> Maybe, but since I am unaware of the algorithm, so I am not sure if it's the right candidate.

Well, somebody else can help out then.

Thanks btw!